sdk/cosmos/azure_data_cosmos/src/options/mod.rs

```diff
@@ -57,6 +125,54 @@ pub struct ItemOptions<'a> {
     pub enable_content_response_on_write: bool,
 }
 
+impl AsHeaders for ItemOptions<'_> {
+    type Error = Infallible;
```

Comment:
Nice job noticing this in other implementations! The `as_headers` method is specified as fallible (i.e. it returns `Result` because it could fail) so that the caller knows to properly handle any errors. So, the `AsHeaders` trait requires that we specify what the _type_ of that error would be. However, many AsHeaders implementations (including this one) actually _cannot_ fail, so we use the type `Infallible` as the error, which is a type that cannot actually be constructed (sometimes called a "never" type). Rust can optimize around this a bit and save storage space since it knows that in this case there can never actually _be_ an error.